pkg/utils/applicator/applicator_test.go

```diff
@@ -0,0 +1,440 @@
+package applicator
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ctx = context.Background()
+var errFake = errors.New("fake error")
+
+type testObject struct {
+	runtime.Object
+	meta.ObjectMeta
+}
+
+func (t *testObject) DeepCopyObject() runtime.Object {
+	return &testObject{ObjectMeta: *t.ObjectMeta.DeepCopy()}
+}
+
+func (t *testObject) GetObjectKind() schema.ObjectKind {
+	return schema.EmptyObjectKind
+}
+
+type testNoMetaObject struct {
+	runtime.Object
+}
+
+func TestAPIApplicatorPatchingApply(t *testing.T) {
```

Comment:
I think a mocked client is not enough, I hope to see real case using ginkgo(suit test) that can interact with K8s apiserver.

The tese cases should contain:

1. Deployment as Component(workload) , do strategyMerge patch。
    - normal field patch(add/update/delete)
    - array field patch(add/update/delete)
2. CRD as Component(workload) , do jsonMerge patch。
       - normal field patch(add/update/delete)
       - array field patch(add/update/delete)

Reply:
Sure, I will cover these cases.